### PR TITLE
File cleaner

### PR DESCRIPTION
**Pull request by wolverian**
Delete files that are older than one day (I didn't find a way to parametrise this in a way that PostgreSQL would understand) and are still in draft mode (i.e. not finalised).

### CODE REVIEW

File: src/liiteri/file_cleaner.clj [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+            [org.httpkit.client :as http]
+            [ring.util.http-response :as response]
+            [taoensso.timbre :as log]))
+
+(defn- clean-files [db storage-engine config]
+  (log/info "Cleaning files")
+  (jdbc/with-db-transaction [datasource db]
+    (doseq [file (metadata-store/get-old-draft-files db)]
+      (log/info (str "Cleaning file: " (:key file)))
+      (file-store/delete-file (:key file) storage-engine db))))
```
Comment (auramo) [inline src/liiteri/file_cleaner.clj]: Does metadata get updated? Or does the file come up as deleted on the next round as well?

File: src/liiteri/file_cleaner.clj [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+(defrecord FileCleaner [db storage-engine config]
+  component/Lifecycle
+
+  (start [this]
+    (let [poll-interval (get-in config [:file-cleaner :poll-interval-seconds])
+          times         (c/chime-ch (p/periodic-seq (t/now) (t/seconds poll-interval))
+                                    {:ch (a/chan (a/sliding-buffer 1))})]
+      (log/info "Starting file cleaner process")
+      (a/go-loop []
+        (when-let [_ (a/<! times)]
```
Comment (auramo) [inline src/liiteri/file_cleaner.clj]: Is there a cleaner way to do this, or does chime just suck?

**Comment by auramo** [comment]
Good stuff! Few comments...

File: src/liiteri/file_cleaner.clj (hunk 2 above)
Comment (wolverian) [inline src/liiteri/file_cleaner.clj]: I copy pasted this from @elahti's virus scanner, sorry for being lazy. :/

File: src/liiteri/file_cleaner.clj (hunk 1 above)
Comment (wolverian) [inline src/liiteri/file_cleaner.clj]: Yes, it does.

File: src/liiteri/file_cleaner.clj [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+            [clj-time.core :as t]
+            [clj-time.periodic :as p]
+            [com.stuartsierra.component :as component]
+            [liiteri.db.file-metadata-store :as metadata-store]
+            [liiteri.files.file-store :as file-store]
+            [taoensso.timbre :as log]))
+
+(defn- clean-files [db storage-engine config]
+  (log/info "Cleaning files")
+  (jdbc/with-db-transaction [datasource db]
```
Comment (wolverian) [inline src/liiteri/file_cleaner.clj]: Note: This is copy pasta from the virus checker and I don't understand what this macro does. Why isn't the `datasource` binding used?

File: src/liiteri/file_cleaner.clj (hunk 2 above)
Comment (wolverian) [inline src/liiteri/file_cleaner.clj]: So the answer is: I don't know.

**Pull request merged**